### ISSUE TITLE
Need some modifications on data_augmentation.py

### ISSUE THREAD

**Issue by wenmengliu** [veugene/data_tools#4]
Hi Eugene,

Currently, I am using data_tools on the LiTS2017 dataset. I notice that in the `data_augmentation.py` line 154-156 in the 'image_random_transform' function might need some modifications. Because, in the repo, the input_image shape is based on 'tensorflow' (n_w,n_h,n_c) but if the input_image shape is on 'theano'. Using this function directly will cause the error? I mean I set the input_image shape is on 'theano' in (n_c,n_w,n_h). So I think it might need modifications under this case (Actually I changed it when I used, I just wanna ask it whether I did it right or not). Please kindly correct me if I were wrong. 

<img width="344" alt="screen shot 2018-10-09 at 10 59 22 am" src="https://user-images.githubusercontent.com/11565143/46688873-4f9bdf80-cbb3-11e8-8ad5-e9ebb065db8b.png">

Cheers,
Crystal

**Comment by wenmengliu** [comment]
Hi Eugene,

I mean if the image.data_format is tensorflow, it needs to change the order. Sorry I typed it wrongly.  Now I am working on building up the model. Last week, I finished the architectures of U-net and ResNet and then I could understand your paper and your code. When I finished my temporary work, I might discuss some details with you!

Cheers,
Crystal

**Comment by veugene** [comment]
Yes, it looks like the channel axis handling in `image_random_transform` is incomplete and incorrect for non-default channel dims (default is 0). Currently, `image_stack_random_transform` handles arbitrary channel dims correctly.

Handling a non-zero channel dim could either be fixed in `image_random_transform` or removed.

**Comment by veugene** [comment]
Are you using `fcn_maker`? UNet and ResUNet could be created (in 2D or 3D, with many possible modifications) using `fcn_maker.models.assemble_unet` or `fcn_maker.models.assemble_resunet`.

https://github.com/veugene/fcn_maker/tree/pytorch

**Comment by wenmengliu** [comment]
Actually I tried to use them in my model.  But it seems a little bit complicated for me to apply it directly. So I wrote them by myself with a reference to your repo. I might be too new in this field so I failed to use this fcn_maker directly.

**Comment by Gabsha** [comment]
As a side note there is a neat data augmentation lib called [Augmentor](https://github.com/mdbloice/Augmentor) now, but I don't know if it supports 3D operations.

**Comment by veugene** [comment]
Oh, that looks like it got popular. It seems that for elastic deformations, they anchor all image edges whereas we allow all edges and corners to be deformed since we support 'reflect' filling, where the reflected edges and corners are also deformed and sized appropriately for each deformed image. I wonder how much that matters in practice.
